### PR TITLE
Fixes #6679 - patch to make component accessible

### LINKED ISSUE #6679: Rating component is not accessible
According to the WAVE tool (extension available at: https://addons.mozilla.org/en-US/firefox/addon/wave-accessibility-tool/), the `Rating` component is not accessible, i.e. it is not clear what each star represents.

Here is a sketch of patch to make it better:

```diff
diff --git a/src/ui/components/Rating/index.js b/src/ui/components/Rating/index.js
index 24e584192..e6364a81c 100644
--- a/src/ui/components/Rating/index.js
+++ b/src/ui/components/Rating/index.js
@@ -131,6 +131,9 @@ export class RatingBase extends React.Component<InternalProps, StateType> {
       if (hoveringOverStar !== null) {
         isSelected = thisRating <= hoveringOverStar;
       }
+
+      const title = this.renderTitle(rating, readOnly, thisRating);
+
       const props = {
         className: makeClassName('Rating-star', `Rating-rating-${thisRating}`, {
           'Rating-selected-star': isSelected,
@@ -140,7 +143,7 @@ export class RatingBase extends React.Component<InternalProps, StateType> {
         key: `rating-${thisRating}`,
         onClick: undefined,
         onMouseEnter: () => this.onHoverStar(thisRating),
-        title: this.renderTitle(rating, readOnly, thisRating),
+        title,
       };
 
       if (readOnly) {
@@ -151,9 +154,21 @@ export class RatingBase extends React.Component<InternalProps, StateType> {
         props.onClick = this.onSelectRating;
       }
 
+      const id = `Rating-rating-${thisRating}-title`;
+
       return (
         // eslint-disable-next-line react/jsx-key
-        <button type="button" value={thisRating} {...props} />
+        <React.Fragment>
+          <button
+            aria-describedby={id}
+            type="button"
+            value={thisRating}
+            {...props}
+          />
+          <span id={id} className="visually-hidden">
+            {title}
+          </span>
+        </React.Fragment>
       );
     });
   }
```

I believe this would be a very nice improvement.

### PR DESCRIPTION
**Pull request by shristit**

### CODE REVIEW

File: src/ui/components/Rating/index.js [hunk 1]
```diff
@@ -140,7 +143,7 @@ export class RatingBase extends React.Component<InternalProps, StateType> {
         key: `rating-${thisRating}`,
         onClick: undefined,
         onMouseEnter: () => this.onHoverStar(thisRating),
-        title: this.renderTitle(rating, readOnly, thisRating),
+          title
```
Comment (championshuttler) [inline src/ui/components/Rating/index.js]: Can you please remove he extra space and add a comma after `title`

Comment (championshuttler) [inline src/ui/components/Rating/index.js]: Can you please remove the extra space and add a comma after `title` please :)

File: src/ui/components/Rating/index.js [hunk 2]
```diff
@@ -151,9 +154,21 @@ export class RatingBase extends React.Component<InternalProps, StateType> {
...
 
+      const id = `Rating-rating-${thisRating}-title`;
+
       return (
         // eslint-disable-next-line react/jsx-key
-        <button type="button" value={thisRating} {...props} />
+        <React.Fragment>
+          <button
+            aria-describedby={id}
+            type="buttn"
```
Comment (championshuttler) [inline src/ui/components/Rating/index.js]: Nit: Can you please change it to `button` .

File: src/ui/components/Rating/index.js [hunk 3]
```diff
@@ -131,6 +131,9 @@ export class RatingBase extends React.Component<InternalProps, StateType> {
       if (hoveringOverStar !== null) {
         isSelected = thisRating <= hoveringOverStar;
       }
+
+      const title = this.renderTitle(rating, readOnly, thisRating);
+
```
Comment (championshuttler) [inline src/ui/components/Rating/index.js]: You have some trailing space here , can you please remove it.

**Comment by willdurand** [comment]
@shristit hi, do you need help to update your PR?

**Comment by shristit** [comment]
Hi I will be back to it next week, Have college exams

**Comment by willdurand** [comment]
@shristit that looks good, but you need to write some tests now :)

**Comment by willdurand** [comment]
> Hi William , I do not know how to write test case can you help me with that please?

@shristit Yep. So you need to edit this file: https://github.com/mozilla/addons-frontend/blob/c543050abd5b430018c100a1ee7640a32b5231c8/tests/unit/ui/components/TestRating.js

You can see a list of functions starting with `it()`. Each function is a test case. We need to write one test case per "thing" we add/change/fix. In your case, you changed the `button` to not only be a button but to also have an associated `span`.

I think it would make sense to update this test case: https://github.com/mozilla/addons-frontend/blob/c543050abd5b430018c100a1ee7640a32b5231c8/tests/unit/ui/components/TestRating.js#L186-L193

You can make sure the `button` has a prop called `aria-describedby`, with `Rating-rating-${rating}-title` as value (hint: you can use `toHaveProp()`).

You should also look for a `span` withe the same `id`.

**Comment by willdurand** [comment]
Hi @shristit, do you need help? Here is a test case you could write:

```diff
﻿commit 26a3525b53cac41028212874031023718298e7e8
Author: William Durand <will+git@drnd.me>
Date:   Mon Nov 19 13:52:08 2018 +0100

    add test

diff --git a/tests/unit/ui/components/TestRating.js b/tests/unit/ui/components/TestRating.js
index 45f891b96..c57268f50 100644
--- a/tests/unit/ui/components/TestRating.js
+++ b/tests/unit/ui/components/TestRating.js
@@ -192,6 +192,26 @@ describe(__filename, () => {
     });
   });
 
+  it('renders an accessible button for each rating', () => {
+    const root = renderWithEmptyRating();
+    [1, 2, 3, 4, 5].forEach((rating) => {
+      const button = getStar({ root, rating });
+      const id = `Rating-rating-${rating}-title`;
+
+      expect(button).toHaveProp('aria-describedby', id);
+
+      // Each rating should have a `span` along with a button.
+      expect(root.find(`span[id="${id}"]`)).toHaveLength(1);
+      expect(root.find(`span[id="${id}"]`)).toHaveProp(
+        'className',
+        'visually-hidden',
+      );
+      expect(root.find(`span[id="${id}"]`)).toHaveText(
+        `Rate this add-on ${rating} out of 5`,
+      );
+    });
+  });
+
   it('renders an accessible description for null stars', () => {
     const root = renderWithEmptyRating();

```

**Comment by willdurand** [comment]
Awesome, thanks @shristit! Did you understand the test case?

**Comment by caitmuenster** [comment]
Yay, thanks for the patch @shristit! 🎉 Your contribution has been added to our [recognition wiki](https://wiki.mozilla.org/Add-ons/Contribute/Recognition#November_2018). 

Would you be interested in creating an account on [mozillians.org](https://mozillians.org)? I'd be happy to vouch for you.

**Pull request merged**